lib/spack/spack/binary_distribution.py

```diff
@@ -0,0 +1,209 @@
+# "Benedikt Hegner (CERN)"
+
+import os
+import platform
+import yaml
+
+import llnl.util.tty as tty
+from llnl.util.filesystem import mkdirp
+
+from architecture import get_full_system_from_platform
+from spack.util.executable import which
+import spack.cmd
+import spack
+from spack.stage import Stage
+import spack.fetch_strategy as fs
+import spack.relocate
+
+
+def prepare():
+    """
+    Install patchelf as pre-requisite to the
+    required relocation of binary packages
+    """
+    if platform.system() == 'Darwin':
+        return
+    dir = os.getcwd()
+    patchelf_spec = spack.cmd.parse_specs("patchelf", concretize=True)[0]
+    if not spack.install_layout.check_installed(patchelf_spec):
+        patchelf = spack.repo.get(patchelf_spec)
+        patchelf.do_install()
+    os.chdir(dir)
+
+
+def buildinfo_file_name(spec):
+    """
+    Filename of the binary package meta-data file
+    """
+    return os.path.join(spec.prefix, ".spack", "binary_distribution")
+
+
+def read_buildinfo_file(package):
+    """
+    Read buildinfo file
+    """
+    filename = buildinfo_file_name(package)
+    with open(filename, 'r') as inputfile:
+        content = inputfile.read()
+        buildinfo = yaml.load(content)
+    return buildinfo
+
+
+def write_buildinfo_file(spec):
+    """
+    Create a cache file containing information
+    required for the relocation
+    """
+    text_to_relocate   = []
+    binary_to_relocate = []
+    blacklist = (".spack", "man")
```

Comment:
If it saves tons of time it might be worth it 😄.  I actually _haven't_ seen anyone put binaries in `man`, but it does seem like a good idea to be complete.